src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderTransactionHistoryResourceTest.java

```diff
@@ -0,0 +1,338 @@
+        assertThat(response.getBody().size()).isEqualTo(1);
+    }
+
+    @Test
+    public void When_GetOffenderTransactionHistory_And_OneRecordExisting_And_AccountCodeIsREG_And_DifferentDate_Then_ReturnOneItem() {
+
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+        final var httpEntity = createHttpEntity(token, null);
+        final var url = "/api/offenders/{offenderNo}/transaction-history?account_code=cash&from_date=2019-11-17&to_date=2019-11-17";
+
+        final var response = testRestTemplate.exchange(
+                url,
+                HttpMethod.GET,
+                httpEntity,
+                new ParameterizedTypeReference<List<OffenderTransactionHistoryDto>>() {},
+                OFFENDER_NUMBER);
+
+        assertThat(response.getStatusCodeValue()).isEqualTo(HTTP_OK);
+        assertThat(response.getBody()).isInstanceOf(List.class);
+        assertThat(response.getBody().size()).isEqualTo(1);
+    }
+
+    @Test
+    public void When_GetOffenderTransactionHistory_And_OneRecordExisting_And_AccountCodeIsMissing_And_DifferentDate_Then_ReturnOneItem() {
+
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+        final var httpEntity = createHttpEntity(token, null);
+        final var url = "/api/offenders/{offenderNo}/transaction-history?from_date=2019-11-17&to_date=2019-11-17";
+
+        final var response = testRestTemplate.exchange(
+                url,
+                HttpMethod.GET,
+                httpEntity,
+                new ParameterizedTypeReference<List<OffenderTransactionHistoryDto>>() {},
+                OFFENDER_NUMBER);
+
+        assertThat(response.getStatusCodeValue()).isEqualTo(HTTP_OK);
+        assertThat(response.getBody()).isInstanceOf(List.class);
+        assertThat(response.getBody().size()).isEqualTo(1);
+    }
+
+    @Test
+    public void When_GetOffenderTransactionHistory_And_FromDateIsMissing_And_ToDateIsNotToday_Then_ReturnError() {
+
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+        final var httpEntity = createHttpEntity(token, null);
+        final var url = "/api/offenders/{offenderNo}/transaction-history?to_date=2019-11-17";
+
+        final var response = testRestTemplate.exchange(
+                url,
+                HttpMethod.GET,
+                httpEntity,
+                new ParameterizedTypeReference<ErrorResponse>() {},
+                OFFENDER_NUMBER);
+
+        assertThat(response.getBody().getDeveloperMessage()).isEqualTo("toDate can't be before fromDate");
+        assertThat(response.getBody().getStatus().intValue()).isEqualTo(HTTP_BAD_REQ);
+        assertThat(response.getBody().getUserMessage()).isEqualTo("toDate can't be before fromDate");
+    }
+
+    @Test
+    public void When_GetOffenderTransactionHistory_And_FromDateIsMissing_And_ToDateIsToday_Then_NoErrorIsThrown() {
+
```

Comment:
Is it worth verifying that the repository was called with the correct default date? Might be easier to test if that default date logic lived in the service.